src/_pytest/compat.py

```diff
@@ -314,15 +314,24 @@ def safe_isclass(obj: object) -> bool:
 
 
 def get_user_id() -> int | None:
-    """Return the current user id, or None if we cannot get it reliably on the current platform."""
-    # win32 does not have a getuid() function.
-    # On Emscripten, getuid() is a stub that always returns 0.
-    if sys.platform in ("win32", "emscripten"):
+    """Return the current process's real user id or None if it cannot be
+    retrieved reliably.
+
+    :return: The user id or None
+    """
+    # mypy follows the version and platform checking expectation of PEP 484:
+    # https://mypy.readthedocs.io/en/stable/common_issues.html?highlight=platform#python-version-and-system-platform-checks
+    # Containment checks are too complex for mypy v1.5.0 and cause failure.
+    if sys.platform == "win32" or sys.platform == "emscripten":
+        # win32 does not have a getuid() function.
+        # Emscripten has a return 0 stub.
         return None
-    # getuid shouldn't fail, but cpython defines such a case.
-    # Let's hope for the best.
-    uid = os.getuid()
-    return uid if uid != -1 else None
+    else:
+        # On other platforms, a return value of -1 is assumed to indicate that
+        # the current process's real user id cannot be retrieved reliably.
+        UNRELIABLE = -1
+        uid = os.getuid()
+        return uid if uid != UNRELIABLE else None
```

Comment:
As I commented, I think `ERROR` would be a better name.

```suggestion
        ERROR = -1
        uid = os.getuid()
        return uid if uid != ERROR else None
```